TST: pytest-xdist failures

As part of the effort to add multi-process testing support to `runtests.py`, I encounted another issue that seems related to `pykokkos` import machinery and `main` blocks, even if I only tell `pytest-xdist` to use a single process:

<details>
<summary> `runtests.py` diff below the fold </summary>

```diff
diff --git a/runtests.py b/runtests.py
index b1fbe36..d8afc7d 100644
--- a/runtests.py
+++ b/runtests.py
@@ -1,3 +1,4 @@
+import argparse
 import os
 import shutil
 import pytest
@@ -9,6 +10,19 @@ cwd = os.getcwd()
 shutil.rmtree(os.path.join(cwd, "pk_cpp"),
               ignore_errors=True)
 
+
+# try to support command line arguments to
+# runtests.py that mirror direct usage of
+# pytest
+pytest_args = []
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-n', '--numprocesses', type=int)
+args = parser.parse_args()
+
+if args.numprocesses:
+    pytest_args.extend(["-n", str(args.numprocesses)])
+
 # force pytest to actually import
 # all the test modules directly
-pytest.main()
+pytest.main(pytest_args)
```

</details>

Sample traceback from the testsuite is:

```
________________________________________________________________________________________________________________________ ERROR collecting tests/test_views.py _________________________________________________________________________________________________________________________
tests/test_views.py:6: in <module>
    import pykokkos as pk
pykokkos/__init__.py:17: in <module>
    defaults: Optional[CompilationDefaults] = runtime_singleton.runtime.compiler.read_defaults()
pykokkos/core/compiler.py:260: in read_defaults
    main: Path = self.get_main_path()
pykokkos/core/compiler.py:162: in get_main_path
    return Path(self.console_main)
E   AttributeError: 'Compiler' object has no attribute 'console_main'

```

This is not likely a high-priority matter, but worth keep an eye on this as `pykokkos` evolves. If the much of the testsuite is already parallel with OpenMP we may not gain too much by distributing it along a second dimension anyway, though I was hoping to test that. I think there's an environment variable that controls the number of openmp threads used by default--not sure how/if that interacts with `pykokkos`, but that variable has caused some pain in the scientific Python community in the past with stuff like OpenBLAS, etc.

--- a/pykokkos/core/compiler.py
+++ b/pykokkos/core/compiler.py
@@ -147,20 +147,6 @@ class Compiler:
 
         return members
 
-    def get_main_path(self) -> Path:
-        """
-        Get the path to the main file
-
-        :returns: a Path object to the main file
-        """
-
-        if hasattr(sys.modules["__main__"], "__file__"):
-            path: str = sys.modules["__main__"].__file__
-            path = path[:-3] # remove the .py extensions
-            return Path(path)
-
-        return Path(self.console_main)
-
     def compile_entity(
         self,
         main: Path,
@@ -257,7 +243,8 @@ class Compiler:
         """
 
         defaults: Optional[CompilationDefaults]
-        main: Path = self.get_main_path()
+        module_setup = ModuleSetup(None, km.get_default_space())
+        main: Path = module_setup.get_main_path()
         file: Path = self.get_defaults_file(main)
 
         try:

--- a/pykokkos/core/module_setup.py
+++ b/pykokkos/core/module_setup.py
@@ -104,14 +104,15 @@ class ModuleSetup:
         self.console_main: str = "pk_console"
 
         self.main: Path = self.get_main_path()
-        self.output_dir: Path = self.get_output_dir(self.main, self.metadata, space)
+        self.output_dir: Optional[Path] = self.get_output_dir(self.main, self.metadata, space)
 
-        self.path: str = os.path.join(self.output_dir, self.module_file)
-        self.name: str = self.path.replace("/", "_")
-        self.name: str = self.name.replace("-", "_")
-        self.name: str = self.name.replace(".", "_")
+        if self.output_dir is not None:
+            self.path: str = os.path.join(self.output_dir, self.module_file)
+            self.name: str = self.path.replace("/", "_")
+            self.name: str = self.name.replace("-", "_")
+            self.name: str = self.name.replace(".", "_")
 
-    def get_output_dir(self, main: Path, metadata: EntityMetadata, space: ExecutionSpace) -> Path:
+    def get_output_dir(self, main: Path, metadata: EntityMetadata, space: ExecutionSpace) -> Optional[Path]:
         """
         Get the output directory for an execution space
 
@@ -120,6 +121,9 @@ class ModuleSetup:
         :param space: the execution space to compile for
         :returns: the path to the output directory for a specific execution space
         """
+
+        if metadata.path is None:
+            return None
 
         if space is ExecutionSpace.Default:
             space = km.get_default_space()

